yarn/src/main/scala/org/apache/spark/deploy/yarn/security/CredentialUpdater.scala

```diff
@@ -107,8 +110,16 @@ private[spark] class ExecutorDelegationTokenUpdater(
     }
   }
 
+  private def getTimeOfNextUpdateFromFileName(credentialsPath: Path): Long = {
+    val name = credentialsPath.getName
```

Comment:
IIUC credential file name is combined by `credentials-uuid-timestamp-integer`, uuid has `-` in it. Maybe it is safer to use the current way.